tachyon/crypto/hashes/sponge/poseidon/poseidon.h

```diff
@@ -46,7 +46,42 @@ struct PoseidonSponge final : public PoseidonSpongeBase<PoseidonSponge<F>> {
     state.elements += config.ark.row(round_number);
   }
 
-  void ApplyMix(bool) { state.elements = config.mds * state.elements; }
+  void ApplyMix(bool) {
+    // NOTE (chokobole): Eigen matrix multiplication has a computational
```

Comment:
Could you add a link to the Eigen matrix multiplication code please? Benchmarks would be good for sure but maybe code is enough.